Example/FirestoreSample/FirestoreSample/Views/FavouriteFruitsView.swift

```diff
@@ -0,0 +1,72 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+import FirebaseFirestoreSwift
+
+private struct Fruit: Codable, Identifiable, Equatable {
+  @DocumentID var id: String?
+  var name: String
+  var isFavourite: Bool
+}
+
+struct FavouriteFruitsView: View {
+  @FirestoreQuery(
+    collectionPath: "fruits",
+    predicates: [
+      .where("isFavourite", isEqualTo: true),
+    ]
+  ) fileprivate var fruitResults: Result<[Fruit], Error>
+
+  @State var showOnlyFavourites = true
+
+  var body: some View {
+    if case let .success(fruits) = fruitResults {
```

Comment:
this is great!

At least in the newer Xcodes, you can use a switch statement with implicit returns for some elegant approaches!
```swift
var body: some View {
      switch fruitResults {
      case .success(fruits):
        FruitsList(fruits)
      case .failure(error):
        EmptyView(error)
      }
}
```

Reply:
That's a great suggestion - my goal is to strike a balance between making the code easy to understand for casual reader who are looking for an example how to use this in their own apps (i.e. potentially just wrapping their existing code in a conditional statement), and making it as elegant as possible (as one might do for a production app).

People new to SwiftUI might not be immediately familiar with the idea of extracting views into sub views, properties, or view builders.